Snowflake.Data.Tests/IntegrationTests/StructuredTypesIT.cs

```diff
@@ -0,0 +1,1471 @@
+                    Assert.AreEqual(1, map.Count);
+                    CollectionAssert.AreEqual(new string[] {"a"}, map.Keys);
+                    CollectionAssert.AreEqual(new string[] {"b", "c"}, map["a"]);
+                }
+            }
+        }
+
+        [Test]
+        public void TestSelectMapOfMaps()
+        {
+            // arrange
+            using (var connection = new SnowflakeDbConnection(ConnectionString))
+            {
+                connection.Open();
+                using (var command = connection.CreateCommand())
+                {
+                    EnableStructuredTypes(connection);
+                    var mapAsSFString = "OBJECT_CONSTRUCT('a', OBJECT_CONSTRUCT('b', 'c'))::MAP(TEXT, MAP(TEXT, TEXT))";
+                    command.CommandText = $"SELECT {mapAsSFString}";
+                    var reader = (SnowflakeDbDataReader)command.ExecuteReader();
+                    Assert.IsTrue(reader.Read());
+
+                    // act
+                    var map = reader.GetMap<string, Dictionary<string, string>>(0);
+
+                    // assert
+                    Assert.AreEqual(1, map.Count);
+                    var nestedMap = map["a"];
+                    Assert.AreEqual(1, nestedMap.Count);
+                    Assert.AreEqual("c", nestedMap["b"]);
+                }
+            }
+        }
+
+        [Test]
+        public void TestSelectAllUnstructuredTypesObject()
+        {
+            // arrange
+            using (var connection = new SnowflakeDbConnection(ConnectionString))
+            {
+                connection.Open();
+                using (var command = connection.CreateCommand())
+                {
+                    EnableStructuredTypes(connection);
+                    var timeZone = GetTimeZone(connection);
+                    var expectedOffset = timeZone.GetUtcOffset(DateTime.Parse("2024-07-11 14:20:05"));
+                    var expectedOffsetString = ToOffsetString(expectedOffset);
+                    var allTypesObjectAsSFString = @"OBJECT_CONSTRUCT(
+                        'StringValue', 'abc',
+                        'CharValue', 'x',
+                        'ByteValue', 15,
+                        'SByteValue', -14,
+                        'ShortValue', 1200,
+                        'UShortValue', 65000,
+                        'IntValue', 150150,
+                        'UIntValue', 151151,
+                        'LongValue', 9111222333444555666,
+                        'ULongValue', 9111222333444555666,
+                        'FloatValue', 1.23,
+                        'DoubleValue', 1.23,
+                        'DecimalValue', 1.23,
+                        'BooleanValue', true,
+                        'GuidValue', '57af59a1-f010-450a-8c37-8fdc78e6ee93',
+                        'DateTimeValue', '2024-07-11 14:20:05'::TIMESTAMP_NTZ,
+                        'DateTimeOffsetValue', '2024-07-11 14:20:05'::TIMESTAMP_LTZ,
```

Comment:
What about TZ and DATE?

Reply:
see TestSelectDateTime, TestSelectDateTimeOffset